testsuite/pytests/test_spatial/test_ntree_split.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# test_ntree_split.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test cornercase when splitting the ``ntree``.
+
+This test connects spatial populations where the positions of nodes are
+defined such that they cause roundoff errors when the ``ntree`` is split.
+"""
+
+import nest
+import numpy as np
+
+
+def test_ntree_split():
+    """
+    Test that ``ntree`` split does not fail when node positions cause roundoff error.
+    """
+
+    # Generate positions. The parameters pre_n_x and r should be defined such
+    # that we get roundoff errors. pre_n_x must be larger than 100 to make the
+    # ntree split.
+    pre_n_x = 110
+    r = 0.45
+
+    low_xy = -r
+    high_xy = r
+    dx = (high_xy - low_xy) / (pre_n_x - 1)
+    xs = np.arange(low_xy, high_xy + dx, dx)
+    y = 0.0
+    z = 0.0
+
+    positions = [[x, y, z] for x in xs]
```

Comment:
I don't like `arange()` because there could be rare roundoff problems. I'd suggest to simplify this to
```suggestion
    # Generate positions. The values used here have cause rounding errors previously.
    # previously. We need to have more than 100 position to force the ntree to split.
    positions = [[x, 0.0, 0.0] for x in np.linspace(-0.45, 0.45, 110)]
```